From 1133fc9c596961f1c6c77d8139404d7f062dec11 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 13 Apr 2024 16:46:52 +0200
Subject: [PATCH 4/4] docs: kernel_include.py: Fix build with docutils 0.21.1

Recent docutils removed `reprunicode`, which was used to remove the
leading `u` from the repr output of Python 2's `unicode` type. For
Python 3, `reprunicode` was just an alias for `str`.

Signed-off-by: Jan Alexander Steffens (heftig) <heftig@archlinux.org>
---
 Documentation/sphinx/kernel_include.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Documentation/sphinx/kernel_include.py b/Documentation/sphinx/kernel_include.py
index abe768088..4ba03e538 100755
--- a/Documentation/sphinx/kernel_include.py
+++ b/Documentation/sphinx/kernel_include.py
@@ -97,7 +97,7 @@ class KernelInclude(Include):
         # HINT: this is the only line I had to change / commented out:
         #path = utils.relative_path(None, path)
 
-        path = nodes.reprunicode(path)
+        path = str(path)
         encoding = self.options.get(
             'encoding', self.state.document.settings.input_encoding)
         e_handler=self.state.document.settings.input_encoding_error_handler
-- 
2.45.1

